GTBitsOfGood/umi-feeds-app
# Michael/user login logic

## User Login Logic [FRONTEND]

Changes login redux logic to populate the auth state with the user information from the database.

### How to Test

1. Create a new user in the MongoDB database following the user schema:
```
{
      _id?: string; // the unqiue id assigned to a user. Let Mongo create this when you insert a document without any _id attribute
      name: string; // will split string in the frontend
      email: string;
      businessName: string;
      phoneNumber: number;
      pushTokens: string[];
      isAdmin: boolean;
      auth0AccessToken: string;
      roles: string[];
      pickupAddresses: Address[];
      dishes: Dish[];
      donations: DonationForm[];
}
```
2. Navigate to the [auth0 website](https://auth0.com/) and sign in with the credentials from bitwarden.
3. Find your user profile (or sign up) and find your auth0 access token (if using google oauth it will be in the format `google-oauth2|numbernumbernumber`
4. Add this token under auth0AccessToken for the mock user you created.
5. Try signing in and see if it brings you to the HomeScreen
6. Modify the user in MongoDB to adjust the auth0 token slightly
7. Logout and try logging back in
8. See if the application brings you to the onboarding screens

**NOTE:** This update requires changes to the backend to function properly.  See pull request on the backend repo as well.

### Important Changes

- AuthState is now of type AuthUser.  I tried to ensure this would not break anything (AuthUser has property name instead of firstName and lastName), but I could've missed something.
- I added an onboarding action to the authReducer, because I think it will be useful during user creation to have access to the user's name and jwt for authentication with the backend

### Related Github Issues

- #97

### Related PRs

- There is a PR for the backend repo, but it has not been created yet.  It will have the same name as this pr "Michael/user-login-logic"

### Checklist
